src/garage/experiment/experiment.py

```diff
@@ -234,6 +235,43 @@ def _make_sequential_log_dir(log_dir):
             i += 1
 
 
+def _make_experiment_signature(function):
+    """Generate an ExperimentTemplate's signature from its function.
+
+    Checks that the first parameter is named ctxt and removes it from the
+    signature. Makes all other parameters keyword only.
+
+    Args:
+        function (callable[ExperimentContext, ...]): The wrapped function.
+
+    Returns:
+        inspect.Signature: The signature of the ExperimentTemplate.
+
+    Raises:
+        ValueError: If the wrapped function's first parameter is not 'ctxt'.
+
+    """
+    func_sig = inspect.signature(function)
+    new_params = []
+    is_first_param = True
+    for param in func_sig.parameters.values():
+        if is_first_param:
+            # Don't output it to the experiment params, since it will contain
+            # the context.
+            if param.name != 'ctxt':
+                raise ValueError('Experiment functions should have a first '
+                                 'parameter named ctxt')
+            is_first_param = False
+        else:
+            new_params.append(
+                inspect.Parameter(name=param.name,
+                                  kind=inspect.Parameter.KEYWORD_ONLY,
+                                  default=param.default,
+                                  annotation=param.annotation))
+    return inspect.Signature(new_params,
```

Comment:
What about this
```suggestion
    return inspect.Signature(func_sig.parameters[1:],
```
plus name check on the first parameter? This will saves the extra `new_params` list.

Reply:
They also need to be made keyword only, so this isn't sufficient.